packages/ui/src/components/alert/AlertRoot.tsx

```diff
@@ -0,0 +1,121 @@
+import { forwardRef, PropsWithChildren, useLayoutEffect, useRef, useState } from 'react'
+
+import { Button } from '@components/button'
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+import { Icon, IconNameMap } from '../icon'
+
+const alertVariants = cva('cn-alert', {
+  variants: {
+    theme: {
+      info: 'cn-alert-info',
+      danger: 'cn-alert-danger',
+      warning: 'cn-alert-warning'
+    }
+  },
+  defaultVariants: {
+    theme: 'info'
+  }
+})
+
+const iconMap: Record<NonNullable<VariantProps<typeof alertVariants>['theme']>, keyof typeof IconNameMap> = {
+  info: 'info-circle',
+  danger: 'cross-circle',
+  warning: 'warning-triangle-outline'
+}
+
+export interface AlertRootProps extends PropsWithChildren<VariantProps<typeof alertVariants>> {
+  className?: string
+  dismissible?: boolean
+  onDismiss?: () => void
+  expandable?: boolean
+}
+
+export const AlertRoot = forwardRef<HTMLDivElement, AlertRootProps>(
+  ({ className, theme, children, dismissible, onDismiss, expandable }, ref) => {
+    const MAX_HEIGHT = 60
+
+    const [isVisible, setIsVisible] = useState(true)
+    const [isExpanded, setIsExpanded] = useState(false)
+    const [isOverflowing, setIsOverflowing] = useState(false)
+    const contentRef = useRef<HTMLDivElement>(null)
+
+    const handleDismiss = () => {
+      setIsVisible(false)
+      onDismiss?.()
+    }
+
+    const iconName: keyof typeof IconNameMap = iconMap[theme ?? 'info']
+
+    const toggleExpand = () => setIsExpanded(prev => !prev)
+
+    const shouldShowButton = expandable && isOverflowing
+
+    useLayoutEffect(() => {
+      if (!expandable || !contentRef.current) return
+
+      const el = contentRef.current
+      setIsOverflowing(el.scrollHeight > MAX_HEIGHT)
+    }, [children, expandable])
+
+    if (!isVisible) return null
+
+    return (
+      <div ref={ref} role="alert" className={cn(alertVariants({ theme }), className)}>
+        {dismissible && (
+          <Button
+            className="cn-alert-close-button"
+            onClick={handleDismiss}
+            type="button"
+            variant="transparent"
+            size="sm"
+            iconOnly
+          >
+            <Icon name="close" size={14} />
+          </Button>
+        )}
+
+        <Icon className="cn-alert-icon" name={iconName} size={18} />
+
+        <div className="cn-alert-text-wrap">
+          <div
+            className={cn('cn-alert-content-box', {
+              'grid-rows-[1fr]': isExpanded,
+              'grid-rows-[0fr]': !isExpanded && shouldShowButton
+            })}
+          >
+            <div
+              ref={contentRef}
+              className={cn('cn-alert-content', {
+                'min-h-[60px]': shouldShowButton
+              })}
+              role="region"
+              aria-label="Alert content"
+              aria-hidden={expandable ? !isExpanded : undefined}
```

Comment:
By default, the content is accessible with or without this button, but with the `aria-hidden` attribute, it's not accessible when hidden for regular users
We'd better remove this